src/parallel/thread_context.cpp

```diff
@@ -5,6 +5,19 @@
 namespace duckdb {
 
 ThreadContext::ThreadContext(ClientContext &context) : profiler(QueryProfiler::Get(context).IsEnabled()) {
+	const auto &tree_map = QueryProfiler::Get(context).GetTreeMap();
+	//	D_ASSERT(tree_map.empty());
+	if (!tree_map.empty()) {
+		const auto &first = tree_map.begin()->second.get();
+
+		if (first.settings.setting_enabled(TreeNodeSettingsType::OPERATOR_TIMING) ||
+		    first.settings.setting_enabled(TreeNodeSettingsType::CPU_TIME)) {
+			profiler.EnableOperatorTiming();
+		}
+		if (first.settings.setting_enabled(TreeNodeSettingsType::OPERATOR_CARDINALITY)) {
+			profiler.EnableOperatorCardinality();
+		}
+	}
```

Comment:
Where was this happening before, and why are we moving it into the thread context?

Reply:
This wasn't happening before but this is where the `OperatorProfiler` is constructed (`profiler = OperatorProfiler()`).  And since I added the boolean into the `OperatorProfiler` I have to pull them out of the `QueryProfiler` here. But if we remove the booleans and add the map instead that will have to happen here too, but in that case we would take it out of the `ClientContext`.